### PR TITLE
[1LP][RFR] Aligning cloud provision attributes

### PR DESCRIPTION
**Pull request by lkhomenk**
Purpose or Intent
=================
Needs Yaml update 1st - MR - 441
{{pytest: -v --long-running  cfme/tests/cloud/test_provisioning.py cfme/tests/services/test_cloud_service_catalogs.py  --use-provider azure --use-provider gce_central}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by lkhomenk** [comment]
Azure PRT passes 100$
RHOS11 - UI provision won't succeed with m1.small - env issues. and smaller flavor are filtered in UI. although REST provision can use it
EC2 - wont make it during the week - we can't get list of flavor/templates - Rate Exceeded 
GCE - REST provision fails on 5.9 - BZ is opened / on 5.8 REST provision fails on teardown. API issue

File: cfme/tests/cloud/test_provisioning.py [hunk 1]
```diff
@@ -327,7 +307,9 @@ def test_manual_placement_using_rest(
 
     # find out cloud network
     assert provider_data['cloud_networks']
-    cloud_network_name = provisioning.get('cloud_network')
+    cloud_network_name = provisioning.get('cloud_network').strip()
+    if ' ' in cloud_network_name and provider.one_of(EC2Provider):
+        cloud_network_name = cloud_network_name.split(' ')[0].strip()
```
Comment (quarckster) [inline cfme/tests/cloud/test_provisioning.py]: **REQUIRED**
```python
cloud_network_name = cloud_network_name.split()[0].strip()
```

**Comment by mshriver** [comment]
@lkhomenk can you comment on the latest PRT results? lots of provisioning failures that don't quite seem to match your description of the failure reasons from 5 days ago.  With such a significant change to the instance fixture's provisioning attributes recursive update, I would expect to see better test results.

**Comment by mshriver** [comment]
@lkhomenk travis failure from a bad import that I'm not seeing on master. Looks like this branch is 83 commits behind upstream/master

**Comment by lkhomenk** [comment]
rebased. azure has passed PRT. restarting for multiple providers

File: cfme/tests/services/test_cloud_service_catalogs.py [hunk 2]
```diff
@@ -24,53 +27,51 @@
...
-    # GCE accepts only lowercase letters in VM name
-    vm_name = vm_name.lower()
-    request.addfinalizer(lambda: cleanup_vm(vm_name + "_0001", provider))
+    wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
+    request.addfinalizer(lambda: cleanup_vm(vm_name + "0001", provider))
     image = provisioning['image']['name']
-    item_name = fauxfactory.gen_alphanumeric()
-    provisioning_data = {
-        'catalog': {'vm_name': vm_name,
+    item_name = provider.name + '-service-' + fauxfactory.gen_alphanumeric()
```
Comment (quarckster) [inline cfme/tests/services/test_cloud_service_catalogs.py]: **REQUIRED**
Please use `format`

File: cfme/tests/services/test_cloud_service_catalogs.py [hunk 3]
```diff
@@ -80,7 +81,8 @@ def test_cloud_catalog_item(appliance, setup_provider, provider, dialog, catalog
                                dialog=dialog,
                                catalog_name=image,
                                provider=provider,
-                               prov_data=provisioning_data)
+                               prov_data=inst_args)
+    request.addfinalizer(lambda: catalog_item.delete())
```
Comment (quarckster) [inline cfme/tests/services/test_cloud_service_catalogs.py]: **OPTIONAL**
```python
request.addfinalizer(catalog_item.delete)
```

File: cfme/tests/services/test_cloud_service_catalogs.py [hunk 4]
```diff
@@ -24,53 +27,51 @@
...
     Metadata:
         test_flag: provision
     """
-    # azure accepts only 15 chars vm name
-    vm_name = 'test{}'.format(fauxfactory.gen_string('alphanumeric', 5))
-    # GCE accepts only lowercase letters in VM name
-    vm_name = vm_name.lower()
-    request.addfinalizer(lambda: cleanup_vm(vm_name + "_0001", provider))
+    wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
+    request.addfinalizer(lambda: cleanup_vm(vm_name + "0001", provider))
```
Comment (quarckster) [inline cfme/tests/services/test_cloud_service_catalogs.py]: **REQUIRED**
Please use `format`

File: cfme/tests/services/test_cloud_service_catalogs.py [hunk 5]
```diff
@@ -82,7 +82,7 @@ def test_cloud_catalog_item(appliance, vm_name, setup_provider, provider, dialog
                                catalog_name=image,
                                provider=provider,
                                prov_data=inst_args)
-    request.addfinalizer(lambda: catalog_item.delete())
+    request.addfinalizer(catalog_item.delete())
```
Comment (quarckster) [inline cfme/tests/services/test_cloud_service_catalogs.py]: **REQUIRED**
A callable should be provided to `addfinalizer`
```python
request.addfinalizer(catalog_item.delete)
```

**Comment by mshriver** [comment]
Waiting on @lkhomenk to include update to wrapanapi 2.7.0 before merge

**Pull request merged**